GoogleMapsComponents/DependencyInjectionExtensions.cs

```diff
@@ -1,50 +1,27 @@
-﻿using Microsoft.Extensions.DependencyInjection;
-using System;
-
-namespace GoogleMapsComponents;
-
-public static class DependencyInjectionExtensions
-{
-    internal static BlazorGoogleMapsOptions MapOptions { get; set; } = new BlazorGoogleMapsOptions();
-    public static IServiceCollection AddBlazorGoogleMaps(this IServiceCollection service, Action<BlazorGoogleMapsOptions>? opts = null)
-    {
-        MapOptions = new BlazorGoogleMapsOptions();
-        opts?.Invoke(MapOptions);
-
-        if (MapOptions.UseBootstrapLoader)
-        {
-            if (MapOptions.KeyProvider is null)
-            {
-                throw new ArgumentNullException(nameof(MapOptions.KeyProvider), "Key is required when using bootstrap loader.");
-            }
-
-        }
-
-        return service;
-    }
-}
-
-//All options
-//https://developers.google.com/maps/documentation/javascript/load-maps-js-api#required_parameters
-public class BlazorGoogleMapsOptions
-{
-    /// <summary>
-    /// Is selected option then it uses the bootstrap loader to load the google maps script.
-    /// https://developers.google.com/maps/documentation/javascript/overview#Loading_the_Maps_API
-    /// </summary>
-    public bool UseBootstrapLoader { get; set; }
-
-    /// <summary>
-    /// Default is weekly.
-    /// https://developers.google.com/maps/documentation/javascript/versions
-    /// </summary>
-    public string Version { get; set; } = "weekly";
-
-    /// <summary>
-    /// Comma separated list of libraries to load.
-    /// https://developers.google.com/maps/documentation/javascript/libraries
-    /// </summary>
-    public string? Libraries { get; set; }
-
-    public Func<string>? KeyProvider { get; set; }
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace GoogleMapsComponents;
+
+public static class DependencyInjectionExtensions
+{
+    /// <summary>
+    /// Adds a basic service to provide a google api key for the application when using bootstrap loader method for Google API <para/>
+    /// 
+    /// If implementing your own key service, be sure to register the service as the implementation of <see cref="IBlazorGoogleMapsKeyService"/>,
+    /// so that the <see cref="GoogleMap"/> component can find it. For example instead of this function you might use:
+    /// <code>services.AddScoped&lt;<see cref="IBlazorGoogleMapsKeyService"/>, YourServiceImplmentation&gt;();</code>
+    /// </summary>
+    public static IServiceCollection AddBlazorGoogleMaps(this IServiceCollection services, string key)

```

Comment:
I think this one should be removed. Since by default we will load weekly version of map and many libraries. I think this could couse issues for user since most of them wont need it. If they want to add custom key then should provide other options too.

Reply:
I'm confused by your suggestion here. Don't almost all users need a custom key to match their URL so google can bill them? I thought this would be the most used option since it guarantees that all the libraries needed for different BGM functions are present.

I agree the 'weekly' API release is a somewhat arbitrary choice, and I made it to match the existing default. If we were going for maximum stability of this nuget, then we'd want to put in a static google API version and update the nuget version each time Google updates their build to kind of 'confirm' that they are still compatible for users. But I figured if that were something you or rungwiroon were willing to deal with maintaining that all the time then it'd already be part of the repo and possibly automated with CI/CR that makes automated pull requests about it.

Reply:
So you are saying that by default is good idea to load all libraries. If you think it is ok then we could leave it.

Reply:
If you did want to automate google API versions and sync them with nuget versions, then I'd recommend implementing a bot to automate the pull requests. the Razor.SweetAlert2 team has something like that for their project and it creates pull requests like this one: https://github.com/Basaingeal/Razor.SweetAlert2/pull/2777